### ISSUE TITLE
java.util.NoSuchElementException when calling Generate data action

### ISSUE THREAD

**Issue by pecko** [debrief/limpet#222]
You can reproduce issue as follows:
- create empty dataset
- add some layer
- select layer
- call Generate Data from toolbar

You will get the exception:

```
!ENTRY org.eclipse.ui 4 0 2015-12-29 16:58:05.724
!MESSAGE Unhandled event loop exception
!STACK 0
java.util.NoSuchElementException
    at java.util.ArrayList$Itr.next(ArrayList.java:854)
    at info.limpet.rcp.actions.GenerateDataAction.run(GenerateDataAction.java:30)
    at org.eclipse.jface.action.Action.runWithEvent(Action.java:498)
    at org.eclipse.jface.action.ActionContributionItem.handleWidgetSelection(ActionContributionItem.java:584)
    at org.eclipse.jface.action.ActionContributionItem.access$2(ActionContributionItem.java:501)
    at org.eclipse.jface.action.ActionContributionItem$6.handleEvent(ActionContributionItem.java:452)
    at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:84)
    at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1276)
    at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:3562)
    at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3186)
    at org.eclipse.ui.internal.Workbench.runEventLoop(Workbench.java:2701)
    at org.eclipse.ui.internal.Workbench.runUI(Workbench.java:2665)
    at org.eclipse.ui.internal.Workbench.access$4(Workbench.java:2499)
    at org.eclipse.ui.internal.Workbench$7.run(Workbench.java:679)
    at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:332)
    at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:668)
    at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:149)
    at info.limpet.rcp.product.Application.start(Application.java:112)
    at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
    at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:110)
    at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:79)
    at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:353)
    at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:180)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:497)
    at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:629)
    at org.eclipse.equinox.launcher.Main.basicRun(Main.java:584)
    at org.eclipse.equinox.launcher.Main.run(Main.java:1438)
    at org.eclipse.equinox.launcher.Main.main(Main.java:1414)
```

It is possible to get the same exception when calling Add layer, but I can't find reproducible case.
I will fix the both issues within the PR: https://github.com/debrief/limpet/pull/221

**Comment by pecko** [comment]
Fixed by commit https://github.com/debrief/limpet/commit/a76ad312b508804d5f4531e1df8519977bfcbc15 within the PR: https://github.com/debrief/limpet/pull/221

**Comment by pecko** [comment]
Fixed within #221 . Commit: https://github.com/debrief/limpet/commit/1449140b970bea2c047f0b80c3b1c08508e01993
